### PR TITLE
Brojd/further labels polishing

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does
Polishing labels

## Which issue(s) this PR fixes
https://github.com/grafana/oncall-private/issues/2355
https://github.com/grafana/oncall-private/issues/2305
https://github.com/grafana/oncall-private/issues/2336 

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: grafana-plugin/src/containers/IntegrationLabelsForm/IntegrationLabelsForm.tsx [hunk 1]
```diff
@@ -271,13 +312,19 @@ const CustomLabels = (props: CustomLabelsProps) => {
...
   return (
     <VerticalGroup>
-      <HorizontalGroup spacing="xs" align="flex-start">
-        <Label>Custom labels</Label>
-      </HorizontalGroup>
+      <Text>Dynamic & Static labels ss</Text>
+      <Text type="secondary" size="small">
+        Dynamic: label values are extracted from the alert payload using Jinja.
+        <br />
+        Keys remain static Static: these are not derived from the payload; both key and value are static.
```
[the remark this answers is not in the document]
Comment (brojd) [reply grafana-plugin/src/containers/IntegrationLabelsForm/IntegrationLabelsForm.tsx:319]: Right! Thanks, will update:)

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx [hunk 2]
```diff
@@ -139,7 +140,30 @@ const IntegrationForm = observer((props: IntegrationFormProps) => {
...
               {store.hasFeature(AppFeature.Labels) && (
                 <div className={cx('labels')}>
-                  <Labels ref={labelsRef} errors={errors?.labels} value={data.labels} />
+                  <Labels
+                    ref={labelsRef}
+                    errors={errors?.labels}
+                    value={data.labels}
+                    description={
+                      <>
+                        Labels {id === 'new' ? 'will be' : ''} applied to the integration and inherited by alert groups.
```
Comment (teodosii) [inline grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx:149]: Does this add double spacing if id is not `new` or is it trimmed?

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx [hunk 3]
```diff
@@ -139,7 +140,30 @@ const IntegrationForm = observer((props: IntegrationFormProps) => {
...
+                        <br />
+                        You can modify behaviour in{' '}
+                        {id === 'new' ? (
+                          'Alert group labels'
+                        ) : (
+                          <PluginLink
+                            onClick={() => {
+                              navigateToAlertGroupLabels(id);
```
Comment (teodosii) [inline grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx:157]: Could go just one liner I guess with `onClick={() => navigateToAlertGroupLabels(id)}`

**Review by teodosii — APPROVED** [review]

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx (hunk 3 above)
Comment (brojd) [reply grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx]: done

**Pull request merged**